src/shared/condition.c

```diff
@@ -239,6 +235,44 @@ static int condition_test_kernel_version(Condition *c, char **env) {
         return true;
 }
 
+static int condition_test_version(Condition *c, char **env) {
+        const char *p = ASSERT_PTR(ASSERT_PTR(c)->parameter);
+        _cleanup_free_ char *word = NULL;
+        const char *ver, *s;
+        struct utsname u;
+        int r;
+
+        assert(c);
+        assert(c->type == CONDITION_KERNEL_VERSION || c->type == CONDITION_VERSION);
+
+        r = extract_first_word(&p, &word, " =!<>$", 0);
+        if (r < 0)
+                return log_debug_errno(r, "Failed to parse compare predicate \"%s\": %m", p);
+
+        /* An empty condition is true, but one where the right operand is missing is invalid. */
```

Comment:
why allow empty to be true?

Reply:
There is a test checking that an empty condition evaluates to true

https://github.com/systemd/systemd/blob/main/src/test/test-condition.c#L470-L474